src/client.rs

```diff
@@ -374,7 +374,8 @@ impl Client {
         self.auth_store.read().await.contains_key(registry)
     }
 
-    async fn store_auth_if_needed(&self, registry: &str, auth: &RegistryAuth) {
+    /// Store the authentication information for this registry if it's not already stored in the client
+    pub async fn store_auth_if_needed(&self, registry: &str, auth: &RegistryAuth) {
```

Comment:
This is a nit/ask for your opinion, but what do you think about calling this `store_auth` instead and then documenting that it only stores it if needed? `store_auth_if_needed` seems like a weird public function name to me 😅

Reply:
I'm not sure about this one. There is already a `store_auth` function (it gets called by this one), and I was even thinking about making that one public as well. Because that could also be handy, as there is currently no way to overwrite the auth info if it's once set. And it gets set before it actually checks if the provided credentials even work. Or you might have rolling credentials and need to rotate them. And if you rename this one to `store_auth`, then you need another name for the function again which always overwrites the auth info.

I don't need this use-cases yet (so I only made `store_auth_if_needed` public for now), but might have them in the future (and it's maybe also useful for others to be able to change credentials). So what are your thoughts about also making `store_auth` public, and I think then the current name with the `_if_needed` makes sense if both of them are available publicly.

Reply:
Since the use case you have for `store_auth_if_needed`, let's just make that public as is. With the additional documentation, it should be clear why it is named that way